apis/r/R/SOMADataFrame.R

```diff
@@ -245,6 +245,10 @@ SOMADataFrame <- R6::R6Class(
           is.data.frame(values) || is_arrow_table(values) || is_arrow_record_batch(values)
       )
 
+      # Leave state unmodified
+      # TODO: this issue will automatically go away on https://github.com/single-cell-data/TileDB-SOMA/issues/3059
+      on.exit(self$reopen(mode = "WRITE"))
```

Comment:
If `mode` ever expands beyond `"READ"` and `"WRITE"`, or changes to `"r"`/`"w"` to align with Python, change to capture the array mode and reopen in said mode
```suggestion
      omode <- self$mode()
      on.exit(self$reopen(mode = omode))
```

Reply:
Awesome, thanks @mojaveazure !! :)